src/Table/HeaderCell.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useContext } from "react";
+import { isBreakpointSatisfied } from "./util/breakpoint";
+import ColVisibilityContext from "./util/colVisibilityContext";
+
+export interface HeaderCellProps {
+  children: React.ReactNode;
+
+  textAlign?: "start" | "end" | "center";
+
+  onClick?: (e: React.MouseEvent | React.KeyboardEvent) => void;
+
+  /**
+   * PRIVATE
+   * Prop for internal implementation. Matches the cell
+   * to it's corresponding entry in `colVisibility`
+   */
+  _colIndex?: number;
+}
+
+/**
+ * A cell unique to the table header.
+ * This component renders as a button when an `onClick` handler is passed.
+ *
+ * This component is wrapped in an ARIA live region so the header cell content
+ * is re-announced if it changes as the result of user interaction.
+ */
+const HeaderCell = ({
+  children,
+  textAlign = "start",
+  onClick,
+  _colIndex = 0,
+}: HeaderCellProps) => {
+  const { currentBreakpoint, colVisibility } = useContext(ColVisibilityContext);
+  const minBreakpoint = colVisibility[_colIndex];
+  const isVisible = isBreakpointSatisfied(minBreakpoint, currentBreakpoint);
+  const isButton = typeof onClick === "function";
+
+  if (!isVisible) return null;
+
+  return (
+    <div aria-live="polite">
```

Comment:
I'm thinking this is how it will work:

1. user clicks `HeaderCell` and our `onClick` handler runs
2. The consuming component of `Table` updates and rerenders with the sort order changed for the affected column
3. The `HeaderCell` has new copy and/or sort indicator. If we include an `aria-label` on the sort indicator icon, changes in its value will be announced

tl;dr - click sort action header cell, hear that the sort changed if you're using a screen reader

Reply:
I think eventually we want to include standard props for rendering sorting and filtering indicators, but for now it may be best to compose it in manually before we make too many assumptions